casepro/backend/junebug.py

```diff
@@ -109,9 +109,15 @@ def __init__(self, json_data):
         if remote_language is not None:
             self.language, _, _ = remote_language.partition('_')
         self.name = json_data.get('details').get('name', None)
-        self.addresses = json_data.get('details').get('addresses')
         self.fields = {}
         self.groups = {}
+        addresses = json_data.get('details').get('addresses')
+        self.urns = []
+        for scheme, address in addresses.items():
+            for urn, details in address.items():
+                if 'optedout' in details and details['optedout'] is True:
+                    continue
+                self.urns.append(scheme+":"+urn)
```

Comment:
I'm wondering if we should take in consideration the `default` detail. Say for the `phone` URN type, if there are multiple addresses with the `phone` type, and one of them has the `default=True` detail, then maybe we don't add the others to the contact, only the ones that are default (and if none are default, then we don't add all like usual)